### PR TITLE
Server Side Secure Renegotiation

### PR DESCRIPTION
**Pull request by ejohnstown**
Enable sever side secure renegotiation in a manner similar to the client side.

### CODE REVIEW

File: src/internal.c [hunk 1]
```diff
@@ -21029,7 +21086,10 @@ static int DoSessionTicket(WOLFSSL* ssl, const byte* input, word32* inOutIdx,
         }
         #endif /* WOLFSSL_DTLS */
 
-        /* check for available size */
+        if (IsEncryptionOn(ssl, 1))
+            sendSz += MAX_MSG_EXTRA;
+
+        /* check for avalaible size */
```
Comment (dgarske) [inline src/internal.c]: "available"

Comment (ejohnstown) [reply src/internal.c]: Changes made. Will write up a doc on running secure renegotiation using the example server and client.

File: src/internal.c [hunk 2]
```diff
@@ -10240,6 +10242,9 @@ static int DoCertificateStatus(WOLFSSL* ssl, byte* input, word32* inOutIdx,
     if (ret != 0)
         SendAlert(ssl, alert_fatal, bad_certificate_status_response);
 
+    if (IsEncryptionOn(ssl, 0))
+        *inOutIdx += ssl->keys.padSz;
```
Comment (toddouska) [inline src/internal.c]: Is a size check needed here?

Comment (ejohnstown) [reply src/internal.c]: It is a good idea. For this scenario, it isn't checked in every case. I'll go back and add the checks. (DoHelloRequest is doing the size check for the pad.)

**Pull request merged**